crates/forge/src/init.rs

```diff
@@ -81,19 +89,39 @@ fn extend_gitignore(path: &Path) -> Result<()> {
     Ok(())
 }
 
+trait ScarbCommandExt {
+    fn maybe_set_scarb_path(&mut self, path: Option<impl Into<PathBuf>>) -> &mut Self;
+}
+
+impl ScarbCommandExt for ScarbCommand {
+    fn maybe_set_scarb_path(&mut self, path: Option<impl Into<PathBuf>>) -> &mut Self {
+        if let Some(path) = path {
+            self.scarb_path(path);
+        }
+        self
+    }
+}
+
 pub fn run(project_name: &str) -> Result<()> {
     let project_path = std::env::current_dir()?.join(project_name);
 
-    ScarbCommand::new_with_stdio()
-        .current_dir(std::env::current_dir().context("Failed to get current directory")?)
-        .arg("new")
-        .arg(&project_path)
-        .run()
-        .context("Failed to initialize a new project")?;
+    let scarb = std::env::var("SCARB").ok();
+
+    // this check if command is run by scarb itself by checking if SCARB env exists
+    // this way we will not run `scarb new` again when `scarb new/init --snforge` is used
+    if scarb.is_none() {
+        ScarbCommand::new_with_stdio()
+            .current_dir(std::env::current_dir().context("Failed to get current directory")?)
+            .arg("new")
+            .arg(&project_path)
+            .run()
+            .context("Failed to initialize a new project")?;
+    }
```

Comment:
I am not really sure if we can rely on this? 🤔 
What if someone just sets the env var manually? 
The purpose of this env var is rather general - choose Scarb binary to use with Forge (whatever the usage may be). 
CC @Draggu @Arcticae

Reply:
#2121